test/integration.js

```diff
@@ -49,4 +60,133 @@ describe('integrations', function() {
       vfs.dest(outputBase),
     ], done);
   });
+
+  it('(*nix) sources a directory, creates a symlink and copies it', function(done) {
+    if (isWindows) {
+      this.skip();
+      return;
+    }
+
+    function assert(files) {
+      var symlinkResult = fs.readlinkSync(outputSymlink);
+      var destResult = fs.readlinkSync(outputDirpathSymlink);
+
+      expect(symlinkResult).toEqual(inputDirpath);
+      expect(destResult).toEqual(inputDirpath);
+      expect(files[0].isSymbolic()).toBe(true);
+      expect(files[0].symlink).toEqual(inputDirpath);
+    }
+
+    pipe([
+      vfs.src(inputDirpath),
+      vfs.symlink(symlinkDirpath),
+      vfs.dest(outputDirpath),
+      concat(assert),
+    ], done);
+  });// How does github determine this?
+
+  it('(windows) sources a directory, creates a junction and copies it', function(done) {
+    if (!isWindows) {
+      this.skip();
+      return;
+    }
+
+    function assert(files) {
+      // Junctions add an ending separator
+      var expected = inputDirpath + path.sep;
+      var symlinkResult = fs.readlinkSync(outputSymlink);
+      var destResult = fs.readlinkSync(outputDirpathSymlink);
+
+      expect(symlinkResult).toEqual(expected);
+      expect(destResult).toEqual(expected);
+      expect(files[0].isSymbolic()).toBe(true);
+      expect(files[0].symlink).toEqual(inputDirpath);
+    }
+
+    pipe([
+      vfs.src(inputDirpath),
+      vfs.symlink(symlinkDirpath),
+      vfs.dest(outputDirpath),
+      concat(assert),
+    ], done);
+  });// Comment to hide the inline github comment
+
+  it('(*nix) sources a symlink and copies it', function(done) {
+    if (isWindows) {
+      this.skip();
+      return;
+    }
+
+    fs.mkdirSync(base);
+    fs.mkdirSync(symlinkDirpath);
+    fs.symlinkSync(inputDirpath, outputSymlink);
+
+    function assert(files) {
+      var destResult = fs.readlinkSync(outputDirpathSymlink);
+
+      expect(destResult).toEqual(inputDirpath);
+      expect(files[0].isSymbolic()).toEqual(true);
+      expect(files[0].symlink).toEqual(inputDirpath);
+    }
+
+    pipe([
+      vfs.src(outputSymlink, { resolveSymlinks: false }),
+      vfs.dest(outputDirpath),
+      concat(assert),
+    ], done);
+  });
+
+  it('(windows) sources a directory symlink and copies it', function(done) {
+    if (!isWindows) {
+      this.skip();
+      return;
+    }
+
+    fs.mkdirSync(base);
+    fs.mkdirSync(symlinkDirpath);
+    fs.symlinkSync(inputDirpath, outputSymlink, 'dir');
+
+    function assert(files) {
+      // 'dir' symlinks add an ending separator
```

Comment:
@erikkemperman I guess our assumption about junctions adding the sep was incorrect.

Reply:
@phated Huh. Where did we get that assumption in the first place? Don't remember. Sorry I haven't gotten around to responding to the above...